Fix NullPointerExceptions with Pipeline. (#296)

na-Itms opened the pull request:
* Fix the CommentBuilder by passing a build result that is correctly sanitized by the BuildResultProcessor.
* Sanitize for non-differential builds as well.

kageiit commented on src/main/java/com/uber/jenkins/phabricator/CommentBuilder.java:
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2015 Uber Technologies, Inc.
+// Copyright (c) 2018 Uber Technologies, Inc.
```
please dont change these

kageiit commented on src/main/java/com/uber/jenkins/phabricator/BuildResultProcessor.java:
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2015 Uber
+// Copyright (c) 2018 Uber
```
please dont change these

kageiit commented:
> Reverted copyright date updates (is there any reason why? just curious, as updating the copyright date sounds like a sane thing to do as the file is updated).
> 
> Please let me know if you prefer having a squashed and rebased pull request.

Copyright dates are not supposed to be updated. They are usually set to the inception date of the file. Im not a lawyer however, but this is our oss policy. You can consult legal resources online if you want to dig into more details

kageiit commented:
Needs javadocs as requested before it can be merged

kageiit commented:
Pulled in as https://github.com/na-Itms/phabricator-jenkins-plugin/commit/a7a55a67a4fe703a6174b829bbc6e296f794f7f4

emiro85 commented:
Hey,

I see that the PR was closed but I don't see the commit in master branch. Can you please check? I'm desperately waiting for this fix to activate my pipeline reporting back to Phabricator.

Thanks.

na-Itms commented:
Hello, yes indeed, I do not understand why this and issue #296 were closed.

I am sorry I did not have time to amend my pull request following the review above, but I did so in commit https://github.com/na-Itms/phabricator-jenkins-plugin/commit/16489a6b53916a2ed7eff19ba339cf6fe7dcb9df.

I can open a new pull request for this commit, which addresses the remarks above, is squashed and rebased against current master. Please let me know 🙂

kageiit commented:
feel free to re-open

na-Itms commented:
Opened #312. I do not have the rights to re-open an issue/PR that was closed by a maintainer.
